axum-extra/src/extract/with_rejection.rs

```diff
@@ -0,0 +1,138 @@
+use axum::async_trait;
+use axum::extract::{FromRequest, RequestParts};
+use axum::response::IntoResponse;
+use std::marker::PhantomData;
+use std::ops::{Deref, DerefMut};
+
+/// Extractor for customizing extractor rejections
+///
+/// `WithRejection` wraps another extractor and gives you the result. If the
+/// extraction fails, the `Rejection` is transformed into `R` and returned as a
+/// response
+///
+/// `E` is expected to implement [`FromRequest`]
+///
+/// `R` is expected to implement [`IntoResponse`] and [`From<E::Rejection>`]
+///
+///
+/// # Example
+///
+/// ```rust
+/// use axum::extract::rejection::JsonRejection;
+/// use axum::response::{Response, IntoResponse};
+/// use axum::Json;
+/// use axum_extra::extract::WithRejection;
+/// use serde::Deserialize;
+///
+/// struct MyRejection { /* ... */ }
+///
+/// impl From<JsonRejection> for MyRejection {
+///     fn from(_:JsonRejection) -> MyRejection {
+///         // ...
+///         # todo!()
+///     }
+/// }
+///
+/// impl IntoResponse for MyRejection {
+///     fn into_response(self) -> Response {
+///         // ...
+///         # todo!()
+///     }
+/// }
+/// #[derive(Debug, Deserialize)]
+/// struct Person { /* ... */ }
+///
+/// async fn handler(
+///     // If the `Json` extractor ever fails, `MyRejection` will be sent to the
+///     // client using the `IntoResponse` impl
+///     WithRejection(Json(Person), _): WithRejection<Json<Person>, MyRejection>
+/// ) { /* ... */ }
+/// ```
```

Comment:
We usually add this to ensure `handler` actually implements `Handler`. Just to make sure the docs don't go out of sync.

```suggestion
# let _: Router = Router::new().route("/", get(handler));
/// ```
```